### ISSUE TITLE
OpenAI::Client.new() replaces shared `extra_headers`

### ISSUE THREAD

**Issue by swistaczek** [alexrudall/ruby-openai#307]
Hey,
I just wanted to express my gratitude for your latest open-source contribution. It's truly amazing and I appreciate your efforts immensely - thank you!

I found out that `OpenAI::Client.new()` overwrites any `extra_headers` which were created in the context of shared configuration.

Example:

```ruby
OpenAI.configure do |config|
  config.extra_headers = { 'X-INVESTMENT-FUND' => 'AND-ITS-GONE` }
end

OpenAI.configuration.extra_headers
 => { 'X-INVESTMENT-FUND' => 'AND-ITS-GONE` }

client = OpenAI::Client.new()

OpenAI.configuration.extra_headers
 => { }
```

**Comment by alexrudall** [comment]
That ain't right

**Comment by alexrudall** [comment]
Fixed in [v4.3.2](https://rubygems.org/gems/ruby-openai/versions/4.3.2) - thanks for raising!
